### PR TITLE
Add BranchDayOfWeekOperator

### PR DESCRIPTION
**Pull request by eladkal**
Add new core operator `BranchDayOfWeekOperator`.
Inspired by [DayOfWeekSensor](https://github.com/apache/airflow/blob/master/airflow/sensors/weekday.py#L25) & [BranchPythonOperator](https://github.com/apache/airflow/blob/master/airflow/operators/python.py#L309)

use case: branch workflow based on day of week.

### CODE REVIEW

File: airflow/operators/weekday.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        week_day: Union[str, Iterable[str]],
+        use_task_execution_day: bool = False,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.follow_task_ids_if_true = follow_task_ids_if_true
+        self.follow_task_ids_if_false = follow_task_ids_if_false
+        self.week_day = week_day
+        self.use_task_execution_day = use_task_execution_day
+        self._week_day_num = None
```
Comment (kaxil) [inline airflow/operators/weekday.py:68]: `follow_task_ids_if_true` sounds like it takes a boolean value

maybe `tasks_to_follow_if_true` ?

Comment (eladkal) [reply airflow/operators/weekday.py:68]: @kaxil Seems there is already convention for it. See BranchSQLOperator https://github.com/apache/airflow/blob/master/airflow/operators/sql.py#L494

Comment (kaxil) [reply airflow/operators/weekday.py:68]: aah right, in that case let's keep that

File: tests/operators/test_weekday.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+        with create_session() as session:
+            session.query(DagRun).delete()
+            session.query(TI).delete()
+
+    def setUp(self):
+        super().setUp()
+        self.dag = DAG(
+            "branch_day_of_week_operator_test",
+            default_args={"owner": "airflow", "start_date": DEFAULT_DATE},
+            schedule_interval=INTERVAL,
```
Comment (kaxil) [inline tests/operators/test_weekday.py]:
```suggestion
            "branch_day_of_week_operator_test",
            start_date=DEFAULT_DATE,
            schedule_interval=INTERVAL,
```

File: tests/operators/test_weekday.py [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+
+
+class TestBranchDayOfWeekOperator(unittest.TestCase):
+    """
+    Tests for BranchDayOfWeekOperator
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
```
Comment (kaxil) [inline tests/operators/test_weekday.py]:
```suggestion
    def setUpClass(cls):
```

File: tests/operators/test_weekday.py [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+
+        with create_session() as session:
+            session.query(DagRun).delete()
+            session.query(TI).delete()
+
+    def setUp(self):
+        super().setUp()
```
Comment (kaxil) [inline tests/operators/test_weekday.py]:
```suggestion
```

File: tests/operators/test_weekday.py [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+            "branch_day_of_week_operator_test",
+            default_args={"owner": "airflow", "start_date": DEFAULT_DATE},
+            schedule_interval=INTERVAL,
+        )
+        self.branch_1 = DummyOperator(task_id="branch_1", dag=self.dag)
+        self.branch_2 = DummyOperator(task_id="branch_2", dag=self.dag)
+        self.branch_3 = None
+
+    def tearDown(self):
+        super().tearDown()
```
Comment (kaxil) [inline tests/operators/test_weekday.py]:
```suggestion
```

**Review by kaxil — APPROVED** [review]
Minors suggestions

**Comment by eladkal** [comment]
@kaxil fixed

**Pull request merged**